openhtf/core/test_executor.py

```diff
@@ -146,6 +146,12 @@ def _thread_proc(self):
 
       # Everything is set, set status and begin test execution.
       self._execute_test_phases(phase_exec)
+      
+  def _thread_exception(self, exc_type, exc_val, exc_tb):
+    """When the framework throws an exception, turn it into a failure code."""
+    if self.test_state and not self.test_state.is_finalized:
+      self.test_state.test_record.add_outcome_details(exc_type.__name__, exc_val.message)
+    return True
```

Comment:
This is going to be a departure from the previous logic, which suppressed only in the case of ThreadTerminationError. Is that intentional?